### ISSUE TITLE
Impact Ratio as a new metric

### ISSUE THREAD

**Issue by yegor256** [yegor256/cam#226]
Let's add a series of new metrics:

  * Impact Ratio (IR): 1 divided by the total number of Java files in the entire repository
  * Impact Ratio by Lines of Code (IRLoC): number of lines of code in a file divided by the total number of lines of code in the entire repository
  * Impact Ratio by Commits (IRC): number of Git commits made in a file divided by the total number of Git commits in the entire repository
  * Impact Ratio by Commit Authors (IRCA): number of Git committers who made changes in a file divided by the total number of Git committers in the entire repository

**Comment by zaqbez39me** [comment]
@yegor256 should we count LOC only in `*.java` files for all the metrics here?

**Comment by yegor256** [comment]
@zaqbez39me yes, of course. There are no other files available in the dataset, when metric calculation starts.

**Comment by zaqbez39me** [comment]
@yegor256  Should we split this task on several PRs? I mean, each subtask test and implementation should be in separate PRs, or all tests in first PR and all the code in second PR?

**Comment by yegor256** [comment]
@zaqbez39me definitely, would be better to do this in a number of PRs. I would suggest to start from tests.

**Comment by zaqbez39me** [comment]
@yegor256 In IRLoC, IRC and IRCA we have repo and some file. Do repo and file paths should be provided separately to the script? If yes, should we check that the file is a part of a given repo? If no, should we change dir to parent of the given file and call there all the git commands needed?

**Comment by yegor256** [comment]
@zaqbez39me I think, we can find the "root" directory of the repo by looking for the `.git` location in upper directories, until we find it. WDYT?
